src/praisonai-code/praisonai_code/cli/utils/stdin.py

```
@@ -0,0 +1,72 @@
+        import select
+
+        # Non-blocking check: only read if data is actually available. Without
+        # this, sys.stdin.read() blocks forever in non-TTY environments
+        # (subprocesses, CI/CD, IDE terminals, Docker) where stdin is a pipe
+        # with no EOF.
+        if select.select([sys.stdin], [], [], 0.0)[0]:
+            stdin_content = sys.stdin.read().strip()
+            return stdin_content if stdin_content else None
+    except Exception:
+        # If there's any error reading stdin, ignore it and fall through.
+        pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)